pkg/asset/installconfig/aws/validation.go

```diff
@@ -166,6 +173,126 @@ func validateAMI(ctx context.Context, config *types.InstallConfig) field.ErrorLi
 	return field.ErrorList{field.Required(field.NewPath("platform", "aws", "amiID"), "AMI must be provided")}
 }
 
+// eipAllocationExists checks if a given EIP allocation ID exists.
+func eipAllocationExists(svc ec2iface.EC2API, allocationID string) (bool, error) {
+	input := &ec2.DescribeAddressesInput{
+		AllocationIds: aws.StringSlice([]string{allocationID}),
+	}
+
+	_, err := svc.DescribeAddresses(input)
+	if err != nil {
+		// Handle specific AWS error for non-existent allocation ID
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "InvalidAllocationID.NotFound" {
+			return false, fmt.Errorf("The allocation ID does not exist")
+		}
+		return false, err
+	}
+
+	return true, nil
+}
+
+func validateEIPAllocations(ctx context.Context, meta *Metadata, fldPath *field.Path, config *types.InstallConfig) field.ErrorList {
+	if config.Platform.AWS.EIPAllocations == nil || len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) == 0 {
+		return nil
+	}
+
+	var eipAllocations []awstypes.EIPAllocation
+	allErrs := field.ErrorList{}
+
+	if config.Platform.AWS != nil && config.Platform.AWS.EIPAllocations != nil && len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) == 0 {
+		return nil
+	} else if len(config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer) > 0 {
+		eipAllocations = config.Platform.AWS.EIPAllocations.IngressNetworkLoadBalancer
+	}
+	var allocationIDs []string
+	for _, allocationID := range eipAllocations {
+		allocationIDs = append(allocationIDs, string(allocationID))
+	}
+
+	sess, err := meta.Session(ctx)
+	if err != nil {
+		return append(allErrs, field.Invalid(fldPath, nil, fmt.Sprintf("unable to start a session: %s", err.Error())))
+	}
+
+	client := ec2.New(sess, aws.NewConfig().WithRegion(config.Platform.AWS.Region))
+
+	// Check if eipAllocations exist
```

Comment:
It looks like you call `svc.DescribeAddresses(input)` quite alot - once per `allocationID` in the for loop for `eipAllocationExists`, and once in `getAssociatedEIPs`. It would be better to call `svc.DescribeAddresses(input)` once here and pass in the array to each function.

Reply:
in this case, then I will have to remove eipAllocationExists function as we fetch the error from DescribeAddressesInput.
If possible please show me what you expect. I tried to do this but it looks like the current approach is more better, even for unit testing.

Reply:
I have changed the logic